// nestjs-oauth2-server/dist/infrastructure/typeorm/client.repository.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientRepository = void 0;
const domain_1 = require("../../domain");
const typeorm_1 = require("typeorm");
const common_1 = require("@nestjs/common");
const modules_dto_1 = require("../../domain/modules/modules.dto");
let ClientRepository = class ClientRepository {
    constructor(repository) {
        this.repository = repository;
    }
    async findAll() {
        return await this.repository.find();
    }
    async find(id) {
        const client = await this.repository.findOne(id);
        if (!client) {
            throw domain_1.ClientNotFoundException.withId(id);
        }
        return client;
    }
    async findByClientId(clientId) {
        const client = await this.repository.findOne({
            where: {
                clientId: clientId
            }
        });
        if (!client) {
            throw domain_1.ClientNotFoundException.withClientId(clientId);
        }
        return client;
    }
    async findByName(name) {
        const client = await this.repository.findOne({
            where: {
                name: name
            }
        });
        if (!client) {
            throw domain_1.ClientNotFoundException.withName(name);
        }
        return client;
    }
    async create(client) {
        return await this.repository.save(client);
    }
    async delete(client) {
        client.deletedAt = new Date();
        return await this.repository.save(client);
    }
};
ClientRepository = __decorate([
    common_1.Injectable(),
    __param(0, common_1.Inject(modules_dto_1.CLIENT_PROVIDER)),
    __metadata("design:paramtypes", [typeorm_1.Repository])
], ClientRepository);
exports.ClientRepository = ClientRepository;
