// nestjs-oauth2-server/dist/domain/modules/access-token.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var AccessTokenModule_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccessTokenModule = void 0;
const common_1 = require("@nestjs/common");
const access_token_entity_1 = require("../access-token.entity");
const modules_dto_1 = require("./modules.dto");
let AccessTokenModule = AccessTokenModule_1 = class AccessTokenModule {
    static forRoot(CONNECTION_PROVIDER) {
        return {
            module: AccessTokenModule_1,
            providers: [
                {
                    provide: modules_dto_1.ACCESS_TOKEN_PROVIDER,
                    inject: [CONNECTION_PROVIDER],
                    useFactory: (connection) => connection.getRepository(access_token_entity_1.AccessTokenEntity),
                },
            ],
            exports: [modules_dto_1.ACCESS_TOKEN_PROVIDER]
        };
    }
};
AccessTokenModule = AccessTokenModule_1 = __decorate([
    common_1.Module({})
], AccessTokenModule);
exports.AccessTokenModule = AccessTokenModule;
;
