// nestjs-oauth2-server/dist/domain/modules/client.module.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var ClientModule_1;
Object.defineProperty(exports, "__esModule", { value: true });
exports.ClientModule = void 0;
const common_1 = require("@nestjs/common");
const client_entity_1 = require("../client.entity");
const modules_dto_1 = require("./modules.dto");
let ClientModule = ClientModule_1 = class ClientModule {
    static forRoot(CONNECTION_PROVIDER) {
        return {
            module: ClientModule_1,
            providers: [
                {
                    provide: modules_dto_1.CLIENT_PROVIDER,
                    inject: [CONNECTION_PROVIDER],
                    useFactory: (connection) => connection.getRepository(client_entity_1.ClientEntity),
                },
            ],
            exports: [modules_dto_1.CLIENT_PROVIDER]
        };
    }
};
ClientModule = ClientModule_1 = __decorate([
    common_1.Module({})
], ClientModule);
exports.ClientModule = ClientModule;
;
