// nestjs-oauth2-server/dist/infrastructure/oauth2-grant-strategy/refresh-token.strategy.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
var __param = (this && this.__param) || function (paramIndex, decorator) {
    return function (target, key) { decorator(target, key, paramIndex); }
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RefreshTokenStrategy = void 0;
const strategy_1 = require("../../domain/strategy");
const dto_1 = require("../../ui/dto");
const common_1 = require("@nestjs/common");
const command_1 = require("../../app/command");
const cqrs_1 = require("@nestjs/cqrs");
let RefreshTokenStrategy = class RefreshTokenStrategy {
    /**
     * Constructor
     *
     * @param clientRepository
     * @param accessTokenRepository
     * @param commandBus
     */
    constructor(clientRepository, accessTokenRepository, commandBus) {
        this.clientRepository = clientRepository;
        this.accessTokenRepository = accessTokenRepository;
        this.commandBus = commandBus;
    }
    async validate(request, client) {
        if ((client.clientSecret && client.clientSecret !== request.clientSecret) ||
            client.deletedAt !== null ||
            !client.grants.includes(request.grantType)) {
            return false;
        }
        return true;
    }
    async getOauth2Response(request, client) {
        const expiredToken = await this.accessTokenRepository.findByRefreshToken(request.refreshToken);
        if (expiredToken.refreshTokenExpiresAt < new Date(Date.now()) || expiredToken.client.clientId !== client.clientId) {
            throw new common_1.UnauthorizedException("You are not allowed to access the given resource");
        }
        // Create a new AccessToken
        const exp = (Date.now() + expiredToken.client.accessTokenLifetime * 1000) / 1000;
        const iat = Date.now() / 1000;
        const accessToken = await this.commandBus.execute(new command_1.CreateAccessTokenCommand(expiredToken.client.id, expiredToken.scope, exp, iat, {
            clientId: expiredToken.client.clientId,
            clientSecret: expiredToken.client.clientSecret,
            exp,
            iat,
            scopes: JSON.parse(expiredToken.scope),
        }, (expiredToken.userId !== null) ? expiredToken.userId : undefined));
        return new dto_1.OAuth2Response(accessToken.accessToken, accessToken.refreshToken, ~~((accessToken.accessTokenExpiresAt.getTime() - Date.now()) / 1000), ~~((accessToken.refreshTokenExpiresAt.getTime() - Date.now()) / 1000));
    }
};
RefreshTokenStrategy = __decorate([
    strategy_1.Oauth2GrantStrategy('refresh_token'),
    __param(0, common_1.Inject('ClientRepositoryInterface')),
    __param(1, common_1.Inject('AccessTokenRepositoryInterface')),
    __metadata("design:paramtypes", [Object, Object, cqrs_1.CommandBus])
], RefreshTokenStrategy);
exports.RefreshTokenStrategy = RefreshTokenStrategy;
